front/migrations/20241211_parents_migrator.ts

```diff
@@ -190,7 +190,61 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
   },
   google_drive: null,
   microsoft: null,
-  github: null,
+  github: {
+    transformer: (nodeId, parents) => {
+      const repoId = parents[parents.length - 1];
+      // case where we are already good
+      if (repoId.startsWith("github-repository")) {
+        return { parents, parentId: parents[1] };
+      }
+      if (parents.length < 3) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Invalid Github parents"
+        );
+      }
+      assert(/^\d+$/.test(repoId), `Invalid repoId: ${repoId}`);
+
+      if (nodeId.startsWith("github-issue-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-issues-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-discussion-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-discussions-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-code")) {
+        // github-code-${repoId}-file-xxx, github-code-${repoId}-dir-xxx, ..., github-code-${repoId}, ${repoId}
+        return {
+          parents: [
+            ...parents,
+            ...parents.slice(1, -2).reverse(),
```

Comment:
here (and for above transforms) it is assumed that github parents all have "correct' form, right? no black sheep

Reply:
I double the parents here, so at least I replicate the old parents (handling the reverse in the transform)

Reply:
sorry for the unclear message; I meant all parents start with self reference and finish with code & repo nodes

Reply:
but yeah in this one in particular I assume that parents are ok (in the other transforms I reconstruct the new parents with the node ID)

Reply:
just read your second comment, yes I do. Adding a warning and ignoring the documents where nodeId !== parents[0], will grep on this warning

Reply:
As for making sure they end with the repo this is caught by the regex I believe